src/test/java/org/generationcp/middleware/brapi/GermplasmListServiceBrapiImplTest.java

```diff
@@ -0,0 +1,85 @@
+package org.generationcp.middleware.brapi;
+
+import org.apache.commons.lang3.RandomStringUtils;
+import org.generationcp.middleware.IntegrationTestBase;
+import org.generationcp.middleware.api.brapi.GermplasmListServiceBrapi;
+import org.generationcp.middleware.api.brapi.v2.germplasm.ExternalReferenceDTO;
+import org.generationcp.middleware.data.initializer.GermplasmListTestDataInitializer;
+import org.generationcp.middleware.domain.search_request.brapi.v2.GermplasmListSearchRequestDTO;
+import org.generationcp.middleware.manager.DaoFactory;
+import org.generationcp.middleware.manager.api.GermplasmListManager;
+import org.generationcp.middleware.pojos.GermplasmList;
+import org.generationcp.middleware.pojos.GermplasmListExternalReference;
+import org.generationcp.middleware.service.api.GermplasmListDTO;
+import org.junit.Assert;
+import org.junit.Before;
+import org.junit.Test;
+import org.springframework.beans.factory.annotation.Autowired;
+
+import java.util.List;
+
+public class GermplasmListServiceBrapiImplTest extends IntegrationTestBase {
+
+	private final String LIST_NAME = RandomStringUtils.randomAlphanumeric(10);
+	private final String LIST_DESCRIPTION = RandomStringUtils.randomAlphanumeric(10);
+	private static final String PROGRAM_UUID = RandomStringUtils.randomAlphanumeric(10);
+	private Integer userId;
+	private Integer germplasmListId;
+
+	@Autowired
+	GermplasmListServiceBrapi germplasmListServiceBrapi;
+
+	@Autowired
+	private GermplasmListManager germplasmListManager;
+
+	private DaoFactory daoFactory;
+	private GermplasmListTestDataInitializer germplasmListTestDataInitializer;
+
+
+	@Before
+	public void setUp() {
+		this.daoFactory = new DaoFactory(this.sessionProvder);
+		this.userId = this.findAdminUser();
+		this.germplasmListTestDataInitializer = new GermplasmListTestDataInitializer();
+		final GermplasmList germplasmList = this.germplasmListTestDataInitializer
+			.createGermplasmList(LIST_NAME, this.userId, LIST_DESCRIPTION, null, 1,
+				PROGRAM_UUID);
+		this.germplasmListId = this.germplasmListManager.addGermplasmList(germplasmList);
+	}
+
+	@Test
+	public void testCountGermplasmListDTOs() {
+		final GermplasmListSearchRequestDTO requestDTO = new GermplasmListSearchRequestDTO();
+		requestDTO.setListDbId(this.germplasmListId.toString());
+		Assert.assertEquals(1, (int) this.germplasmListServiceBrapi.countGermplasmListDTOs(requestDTO));
+	}
+
+	@Test
+	public void testSearchGermplasmListDTOs() {
```

Comment:
Add more tests for filtering scenarios

Reply:
Hi @darla-leafnode , only the following filters are used in the method being tested: listName, listDbId, externalReferenceID, externalReferenceSource. 

listSource is not supported. As for listType, we're only accepting "germplasm" value which then returns all the germplasm lists regardless of the "listtype" value in the database.